backend/pkg/api/handlers/auth.go

```diff
@@ -779,3 +780,66 @@ func (h *HandlerService) ManageViaApiCheckMiddleware(next http.Handler) http.Han
 		next.ServeHTTP(w, r)
 	})
 }
+
+// middleware check to return if specified dashboard is not archived (and accessible)
+func (h *HandlerService) GetVDBArchivedCheckMiddleware(ctx context.Context, r *http.Request, params ...interface{}) error {
+	dashboardId, err := h.handleDashboardId(ctx, mux.Vars(r)["dashboard_id"])
+	if err != nil {
+		return err
+	}
+	dashboard, err := h.dai.GetValidatorDashboard(ctx, *dashboardId)
+	if err != nil {
+		return err
+	}
+	if dashboard.IsArchived {
+		return newForbiddenErr("dashboard with id %v is archived", dashboardId)
+	}
+	return nil
+}
+
+// allows to return a middleware auth check which excludes specified endpoints under the same subrouter
+func (h *HandlerService) GetMiddlewareExcludedRoutes(middlewareCheckFunc func(context.Context, *http.Request, ...interface{}) error, excludedRoutes []*mux.Route, params interface{}) func(http.Handler) http.Handler {
```

Comment:
This seems very complicated for such a minimal use case, as I don't think we'll need this anywhere else. I'd rather just have separate subrouters with their required middlewares instead of doing this exclusion logic.

Reply:
Agreed it does, but encapsulated like this it's prob not much worse than fine-grained custom subrouter creation.

Also I could imagine making use of it again. Not sure if that's good examples, but here are some random thoughts:
1. Maybe we want to introduce in-depth premium-only endpoints (hard to think of a good example.. `/{dashboard_id}/name/history` as a stupid one); instead of checking the users premium status in all associated data access methods, we'd just have one single place to manage
2. Or we want to exclude a number of subroutes for mobile access for some reason, like `/{dashboard_id}/heatmap` because that data can't be presented anyways because the screen is too small or sth.

I can remove it, lmk if you like that version more. But will bookmark this for later :stuck_out_tongue: